### ISSUE TITLE
Reasonable Loss Values ?

### ISSUE THREAD

**Issue by yunusstalha** [Qinyu-Allen-Zhao/Arinar#3]
In GMM head, you guys are using negative log-likelihood loss. What value range is reasonable ? Could you share some loss plots if possible?
best

**Comment by Qinyu-Allen-Zhao** [comment]
Thank you for your interest!

When calculating the NLL loss, I omit a constant term from the loss function. Please see [this issue](https://github.com/Qinyu-Allen-Zhao/Arinar/issues/1) for more information. So the loss value typically starts around 5~10, gradually decreases, and eventually fluctuates around -5 when training on ImageNet.

The figure below shows the loss over the first 70K training steps for the ARINAR model. Although the loss continues to fluctuate around -5 in later stages, the quality of the generated images consistently improves — a common trend observed in training generative models.

Best,
Qinyu

![Image](https://github.com/user-attachments/assets/8469585b-7499-4966-a2be-be34cab990a5)

**Comment by yunusstalha** [comment]
Thanks for this detailed response.

**Comment by yunusstalha** [comment]
What is the right way to train arinar on small scale? I am trying to train on ffhq and random 100 classes in imagenet with the same setup as in your readme.  The loss is like this : 

<img width="873" alt="Image" src="https://github.com/user-attachments/assets/7784f0ec-97e9-472c-b3b4-1c08fff0068a" />

FID is around 400s in both cases and inception is around 1.49.  Do you have any idea on what is wrong? Or any suggestions?

**Comment by Qinyu-Allen-Zhao** [comment]
Hi,

I think the train loss is really weird, because it cannot be so small. These days, I updated the repo and it can support different model heads now, such as diffloss, rectified flow, GMM and ARINAR-GMM. 
https://github.com/Qinyu-Allen-Zhao/Arinar/blob/1047707114ee88ee0c128d0482de54ef0d1d6757/models/mar.py#L106-L147

You can pull the updates and may try a different head. 

Or maybe you can debug the training process and see what's happening in the forward process.

Qinyu
